Финальный проект 7 спринта

TatianaSkuratova opened the pull request.

TatianaSkuratova committed:
Финальный проект 7 спринта

pomegranate282922 commented on src/main/java/AvailableStation.java:
```diff
@@ -0,0 +1,13 @@
+@Setter
+@AllArgsConstructor
+public class AvailableStation {
+    String name;
```
Приватный чтобы не нарушать принцип инкапсуляции 
Нужно исправить

pomegranate282922 commented on src/main/java/Courier.java:
```diff
@@ -0,0 +1,42 @@
+    String password;
+    String firstName;
+
+    public static class Builder{
```
У ломбока есть спец аннотация для билдера 
Можно улучшить

pomegranate282922 commented on src/main/java/Endpoints.java:
```diff
@@ -0,0 +1,9 @@
+public final class Endpoints {
+    public static final String SCOOTER_SERVICE_URI = "https://qa-scooter.praktikum-services.ru/";
```
Отлично что вынесены урлы

pomegranate282922 commented on src/main/java/Order.java:
```diff
@@ -0,0 +1,63 @@
+@Setter
+@Getter
+public class Order {
+    String firstName;
```
Везде приватность нужна

pomegranate282922 commented on src/test/java/TestCourierCreate.java:
```diff
@@ -0,0 +1,123 @@
+import static org.hamcrest.Matchers.equalTo;
+
+public class TestCourierCreate {
+    public static final String SCOOTER_SERVICE_URI = "https://qa-scooter.praktikum-services.ru/";
```
Это тоже лучше вынести в класс с константами

pomegranate282922 commented on src/test/java/TestCourierCreate.java:
```diff
@@ -0,0 +1,123 @@
+    public static final String SCOOTER_SERVICE_URI = "https://qa-scooter.praktikum-services.ru/";
+    public static final String LOGIN = RandomStringUtils.randomAlphabetic(10);
+    public static final String PASSWORD = "123";
+    public static final String FIRST_NAME = RandomStringUtils.randomAlphabetic(15);
```
Отлично что данные рандомизированы

pomegranate282922 commented on src/test/java/TestCourierCreate.java:
```diff
@@ -0,0 +1,123 @@
+public class TestCourierCreate {
+    public static final String SCOOTER_SERVICE_URI = "https://qa-scooter.praktikum-services.ru/";
+    public static final String LOGIN = RandomStringUtils.randomAlphabetic(10);
+    public static final String PASSWORD = "123";
```
Это тоже лучше рандомизировать

pomegranate282922 commented on src/test/java/TestCourierLogin.java:
```diff
@@ -0,0 +1,80 @@
+        ValidatableResponse response = client.login(credential);
+        response.assertThat().body("message", equalTo("Недостаточно данных для входа")).and().statusCode(400);
+    }
+//    @Test
```
Закомментированный код нужно удалить

pomegranate282922 commented on src/test/java/TestGetOrders.java:
```diff
@@ -0,0 +1,77 @@
+
+public class TestGetOrders {
+    ScooterServiceClient client = new ScooterServiceClient();
+    public static final String SCOOTER_SERVICE_URI = "https://qa-scooter.praktikum-services.ru/";
```
Одна и та же константа везде. Лучше не дублировать

pomegranate282922 commented on src/test/java/TestGetOrders.java:
```diff
@@ -0,0 +1,77 @@
+        //создаем заказ
+        String[] blackColor = new String[] {"BLACK"};
+        order = new Order.Builder()
+                .withFirstName("Аня")
```
Тестовые данные нужно вынести из кода самого теста .
Нужно исправить

pomegranate282922 commented on src/main/java/AvailableStation.java:
```diff
@@ -0,0 +1,13 @@
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
```
Отлично что использован ломбок

pomegranate282922 commented on src/main/java/ScooterServiceClient.java:
```diff
@@ -0,0 +1,94 @@
+import io.restassured.response.ValidatableResponse;
+import io.restassured.specification.RequestSpecification;
+import static io.restassured.RestAssured.given;
+public class ScooterServiceClient {
```
Все классы лежат в одной куче. Нужно распределить логически по пакетам.
Нужно исправить

pomegranate282922 reviewed (commented):
Отличная работа! Хорошая структура кода, минимум дублирования, вызовы ручек вынесены из тестов. Все тесты реализованы, используется параметризация.
Здорово, что используется ломбок.
Пара замечаний имеется - поправь, пожалуйста.

TatianaSkuratova committed:
Спасибо за такие подробные замечания. Постаралась все исправить, за исключением аннотации для билдера у ломбока, т.к. по умолчанию он создает приватный конструктор, который я не могу использовать в другом пакете.
Также замечание "Все классы лежат в одной куче." решила исправить не только разделением на пакеты, но также сервисный класс разбила на несколько классов.

pomegranate282922 reviewed (approved):
Отлично! Зачет!
